app/controllers/activitypub/inboxes_controller.rb

```diff
@@ -52,6 +53,20 @@ def upgrade_account
     DeliveryFailureTracker.reset!(signed_request_account.inbox_url)
   end
 
+  def process_collection_synchronization
+    raw_params = request.headers['X-AS-Collection-Synchronization']
+    return if raw_params.blank?
+    return if ENV['DISABLE_FOLLOWERS_SYNCHRONIZATION'] == 'true'
+
+    # Re-using the syntax for signature parameters
+    tree   = SignatureParamsParser.new.parse(raw_params)
+    params = SignatureParamsTransformer.new.apply(tree)
+
+    ActivityPub::PrepareFollowersSynchronizationService.new.call(signed_request_account, params)
+  rescue Parslet::ParseFailed
+    Rails.logger.warn 'Error parsing collection synchronization header'
```

Comment:
Nitpick but I would maybe write "X-AS-Collection-Synchronization" here to be more specific in the error log. Though I also don't remember how we treat errors from "bad input" in other places. Maybe you wouldn't want to fill the warn log level with stuff like this..?

Reply:
I think in general, Mastodon doesn't print enough stuff in logs, but maybe that should be `debug` loglevel rather than `warn`. Updated the text itself.